internal/encryption/keyprovider.go

```diff
@@ -124,6 +125,21 @@ func setupKeyProviders(enc *config.EncryptionConfig, cfgs []config.KeyProviderCo
 	return kpData.hclEvalContext("key_provider"), diags
 }
 
+func keyProvidersStack(stack []config.KeyProviderConfig) ([]string, hcl.Diagnostics) {
```

Comment:
`suggestion`
Would be better for the readability of the code to have this function moved at the bottom of the file. This is more of a personal preference, so that's why I tagged this as `suggestion`.